jena-geosparql/src/main/java/org/apache/jena/geosparql/implementation/access/AccessGeoSPARQL.java

```diff
@@ -229,4 +231,11 @@ private static boolean containsGeoLiterals(Graph graph, Node node) {
             graph.contains(node, Geo.AS_GML_NODE, null);
         return result;
     }
+
+    public static RuntimeException buildException(RuntimeException e) {
+        if (e instanceof QueryCancelledException e2) {
+            return new QueryCancelledException(e2);
+        }
+        return new RuntimeException(e);
+    }
```

Comment:
See `QueryIterFailed`:

```suggestion
    protected static QueryExecException buildException(RuntimeException cause) {
        return switch(cause) {
            case QueryCancelledException ex -> ex;
            case QueryExecException ex -> ex;
            default -> new QueryExecException(cause);
        };
    }
```